src/open-api-helpers.ts

```diff
@@ -514,23 +513,21 @@ export const depictBranded: Depicter<z.ZodBranded<z.ZodTypeAny, any>> = ({
   next,
 }) => next({ schema: schema.unwrap() });
 
-export const defaultSerializer = (schema: z.ZodTypeAny): string =>
-  createHash("sha1").update(JSON.stringify(schema), "utf8").digest("hex");
-
 export const depictLazy: Depicter<z.ZodLazy<z.ZodTypeAny>> = ({
   next,
   schema: lazy,
   serializer: serialize,
-  hasRef,
+  getRef,
   makeRef,
 }): ReferenceObject => {
   const hash = serialize(lazy.schema);
-  if (hasRef(hash)) {
-    return { $ref: `#/components/schemas/${hash}` }; // @todo consider changing it to getRef that return ReferenceObject
-  }
-  const ref = makeRef(hash, {}); // make empty ref first
-  makeRef(hash, next({ schema: lazy.schema })); // update
-  return ref;
+  return (
+    getRef(hash) ||
+    (() => {
+      makeRef(hash, {}); // make empty ref first
+      return makeRef(hash, next({ schema: lazy.schema })); // update
+    })()
```

Comment:
IIFE are cool.
but can I make it even better?
can `makeRef` accept third argument (cb)?

Reply:
I can, but this approach is harder to test